### PR TITLE
Update the publish process

### PR DESCRIPTION
**Pull request by jayudey-wf**
Update CI process for publishing.

### CODE REVIEW

**Comment by aviary2-wf** [comment]
## Security Insights
No security relevant content was detected by automated scans.

## Action Items
* Review PR for [security impact](https://wiki.atl.workiva.net/display/SECURITY/Development+Security+Review+Guidelines); comment "security review required" if needed or unsure
* Verify [`aviary.yaml`](https://wiki.atl.workiva.net/pages/viewpage.action?pageId=42762752#SecurityInsightFramework&Tooling-ConfiguringAviary) coverage of security relevant code
---
_Questions or Comments? Reach out on Slack: #support-infosec._

**Comment by semveraudit-wf** [comment]
### Public API Changes

 ## Issue: No Semver Audit Report Found for Base Commit

**Base commit:** `9013ce9cdc0948fa2b3d6f2ead1c435756bd4823`

This means we couldn't find semver audit data for the commit against which to compare the changes in this PR. Fixing this is a two step process. First, we'll need to generate the semver audit data for the base commit. Then we will have to generate the semver audit data for the head commit on this branch. This can be done by rerunning CI and/or manually running the semver audit tool.

**Note: It may be sufficient to simply merge master into your branch.**

## Step 1: Generate Semver Audit Data for the Base Commit:

You can do this either by rerunning CI on the base commit or by manually running the semver audit tool locally on the base commit.

### Option 1: Rerunning CI

Locate the build associated with the base commit (9013ce9cdc0948fa2b3d6f2ead1c435756bd4823) in [Workiva Build](https://ci.webfilings.com/Workiva/pdfjs_dart) and rerun it.

### Option 2: Run the Semver Audit Tool Manually

```bash
# Navigate to the root of the repo and check out the base commit:
$ git checkout 9013ce9cdc0948fa2b3d6f2ead1c435756bd4823

# If this is a Dart project, run:
$ dart pub global run semver_audit report --repo Workiva/pdfjs_dart --head-commit 9013ce9cdc0948fa2b3d6f2ead1c435756bd4823

# If this is a Go project, run:
$ semver_audit_go report  --repo Workiva/pdfjs_dart --head-commit 9013ce9cdc0948fa2b3d6f2ead1c435756bd4823
```

**Notes:** 
* If frugal files are generated during the build process prior to audit, please generate those locally before running the `semver_audit` command.
* If this is a Dart project, the manual steps above assume you have the Dart 2 SDK installed and activated and the `semver_audit` tool globally installed. See [this Confluence post](https://wiki.atl.workiva.net/questions/104006659/how-do-i-install-dart-2-and-easily-switch-between-dart-versions) to install Dart 2. To globally install the `semver_audit` tool, run the following:

```bash
$ dart pub global activate --hosted-url https://pub.workiva.org semver_audit ^2.0.0
```

## Step 2: Generate Semver Audit Data for this Branch's Head Commit:

You can do this either by rerunning CI on this PR or by manually running the semver audit tool locally against the head commit on this branch.

### Option 1: Rerunning CI

This is often the easiest solution. Find the Workiva Build check on this Pull Request and simply rerun it.

### Option 2: Run the Semver Audit Tool Manually

```bash
# Navigate to the root of the repo and check out the head commit on this branch:
$ git checkout 64863330f3f61a8f05cb0e88caef0e175b07af20

# If this is a Dart project, run:
$ dart pub global run semver_audit report --repo Workiva/pdfjs_dart --base-commit 9013ce9cdc0948fa2b3d6f2ead1c435756bd4823 --head-commit 64863330f3f61a8f05cb0e88caef0e175b07af20 --pull 167

# If this is a Go project, run:
$ semver_audit_go report --repo Workiva/pdfjs_dart --base-commit 9013ce9cdc0948fa2b3d6f2ead1c435756bd4823 --head-commit 64863330f3f61a8f05cb0e88caef0e175b07af20 --pull 167
```

## Still Need Help?

If you are unable to resolve the issue on your own, either tag `@Workiva/developer-experience` in a PR comment or reach out to us in the [#support-frontend-dx](https://slack.com/app_redirect?channel=C03ESR91BNU) Slack channel.


Showing results for 64863330f3f61a8f05cb0e88caef0e175b07af20
> Powered by [semver-audit-service](https://github.com/Workiva/semver-audit-service). Please report any problems by [filing an issue](https://github.com/Workiva/semver-audit-service/issues).
> Reported by the dart semver audit client 2.2.2
[Browse public API](https://semver-audit.wk-dev.wdesk.org/Workiva/pdfjs_dart/64863330f3f61a8f05cb0e88caef0e175b07af20).

**Review by rogerkerr-wk — DISMISSED** [review]

**Review by pauldanner-wk — APPROVED** [review]

**Comment by pauldanner-wk** [comment]
QA +10 :shipit:

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -37,3 +43,8 @@ jobs:
       - name: Check dependencies
         run: dart run dependency_validator
         if: always() && steps.install.outcome == 'success'
+
+      - uses: anchore/sbom-action@v0
```
Comment (ryanelliott-wk) [inline .github/workflows/ci.yaml:47]: why do we need the BOM?

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -8,6 +13,7 @@ on:
   pull_request:
     branches:
       - '**'
+  workflow_call:
```
Comment (ryanelliott-wk) [inline .github/workflows/ci.yaml:16]: this allows publish to run ci?

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (jayudey-wf) [reply .github/workflows/ci.yaml:46]: discussed in real life

**Review by ryanelliott-wk — APPROVED** [review]

**Pull request closed without merging**